tls/test/test_utils.py

```diff
@@ -28,3 +33,160 @@ def test_decode(self, byte, number):
 
 def test_ubint24():
     assert isinstance(UBInt24("test"), _UBInt24)
+
+
+class TestBytesAdapter(object):
+
```

Comment:
We haven't been as strict about this as we should have been, but I really think we should document our tests better. More importantly so _now_, because these are now heading fast towards being less straight-forward (and understandably so!). 

While these tests are awesome (and thanks a lot for adding them! :)), could you please add docstrings to document them? I find https://jml.io/pages/test-docstrings.html to be a handy reference when writing docstrings :)